components/lib/jexl/src/main/java/mozilla/components/lib/jexl/ast/AstNode.kt

```diff
@@ -0,0 +1,143 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.jexl.ast
+
+/**
+ * A node of the abstract syntax tree.
+ *
+ * This class has a lot of properties because it needs to represent all types of nodes (See [AstType]). It should be
+ * possible to make this much simpler by creating a dedicated class for each type.
+ */
+class AstNode(
+    val type: AstType,
+    var value: Any? = null,
+    val operator: String? = null,
```

Comment:
Operator should probably be typed as well e.g. an enum as there's a fixed set of supported operators...

Reply:
This iteration does not yet, but mozjexl supports adding custom operators:
https://github.com/mozilla/mozjexl#api